scripts/migration/migrate_retraction_dates.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Script to migrate retracted registrations so that their date modified is date of retraction."""
+
+import sys
+import logging
+
+from modularodm import Q
+from website.models import Node, NodeLog
+from website.app import init_app
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+
+logger = logging.getLogger(__name__)
+
+
+def do_migration(logs):
+    # ... perform the migration using a list of logs ...
+    for log in logs:
+        registration_id = log.params.get('registration')
+        if registration_id:
+            registration = Node.load(registration_id)
+            if registration.date_modified < log.date:
+                registration.date_modified = log.date
+                registration.save()
+                logger.info('{} date updated to {}'.format(registration, log.date))
+            else:
+                logger.warning('Date modified is more recent than retraction ' + log._id)
+        else:
+            logger.warning('No parent registration found for retraction log ' + log._id)
+
+
+def get_targets():
+    # ... return the list of logs whose registrations we want to migrate ...
+    targets = NodeLog.find(Q('action', 'eq', 'retraction_approved'))
+
+    logger.info('Retractions found: {}'.format(len(targets)))
```

Comment:
Again, not strictly correct, because a 1-1 correlation between retraction approval logs and retracted nodes isn't guaranteed, but not a blocker.